exp/addons/internal/controllers/clusterresourceset_controller.go

```diff
@@ -65,27 +67,52 @@ type ClusterResourceSetReconciler struct {
 	WatchFilterValue string
 }
 
-func (r *ClusterResourceSetReconciler) SetupWithManager(ctx context.Context, mgr ctrl.Manager, options controller.Options) error {
-	err := ctrl.NewControllerManagedBy(mgr).
+func (r *ClusterResourceSetReconciler) SetupWithManager(ctx context.Context, mgr ctrl.Manager, options controller.Options, syncPeriod *time.Duration) error {
+	// Setup a separate cache for the metadata watches to secrets.
+	// This way the watch does not use the LabelSelector defined at the cache which
+	// would filter to secrets having the cluster label, because secrets referred
+	// by ClusterResourceSet are not specific to a single cluster.
+	partialSecretCache, err := cache.New(mgr.GetConfig(), cache.Options{
```

Comment:
I think we have to move this up into main.go

I was running this through the debugger and I noticed that we also have a Secret metadata watch for ExtensionConfig. I think we should use this cache for this one as well

Reply:
Good news is that we only have to pass in the cache here (not the syncPeriod anymore)

Reply:
Please pass in the cache via a struct field "MetadataSecretCache" or something

(just makes it a lot easier to inspect with the debugger at runtime.. :))

Somehow can't find it in the manager currently

Reply:
PTAL

Reply:
> just makes it a lot easier to inspect with the debugger at runtime

i.e. a field on ExtensionConfigReconciler etc, vs just a parameter in SetupWithManager

Reply:
> Good news is that we only have to pass in the cache here (not the syncPeriod anymore)

Sorry for the confusion, we should still use syncPeriod here, I just meant we don't have to pass it down into the reconcile anymore because we're now doing it in main.go